biogeophys/EDSurfaceAlbedoMod.F90

```diff
@@ -195,756 +125,918 @@ subroutine ED_Norman_Radiation (nsites, sites, bc_in, bc_out )
+    real(r8), intent(inout) :: fabi_parb_out(hlm_numSWb)
+    real(r8), intent(inout) :: ftdd_parb_out(hlm_numSWb)
+    real(r8), intent(inout) :: ftid_parb_out(hlm_numSWb)
+    real(r8), intent(inout) :: ftii_parb_out(hlm_numSWb)
+
+    ! Locals
+    ! -----------------------------------------------------------------------------------
+
+    integer  :: radtype, L, ft, j
+    integer  :: iter                                          ! Iteration index
+    integer  :: irep                                          ! Flag to exit iteration loop
+    real(r8) :: sb
+    real(r8) :: error                                         ! Error check
+    real(r8) :: down_rad, up_rad                              ! Iterative solution do Dif_dn and Dif_up
+    real(r8) :: ftweight(nclmax,maxpft,nlevleaf)
+    real(r8) :: k_dir(maxpft)                              ! Direct beam extinction coefficient
+    real(r8) :: tr_dir_z(nclmax,maxpft,nlevleaf)         ! Exponential transmittance of direct beam radiation through a single layer
+    real(r8) :: tr_dif_z(nclmax,maxpft,nlevleaf)         ! Exponential transmittance of diffuse radiation through a single layer
+    real(r8) :: weighted_dir_tr(nclmax)
+    real(r8) :: weighted_fsun(nclmax)
+    real(r8) :: weighted_dif_ratio(nclmax,maxSWb)
+    real(r8) :: weighted_dif_down(nclmax)
+    real(r8) :: weighted_dif_up(nclmax)
+    real(r8) :: refl_dif(nclmax,maxpft,nlevleaf,maxSWb)  ! Term for diffuse radiation reflected by laye
+    real(r8) :: tran_dif(nclmax,maxpft,nlevleaf,maxSWb)  ! Term for diffuse radiation transmitted by layer
+    real(r8) :: dif_ratio(nclmax,maxpft,nlevleaf,maxSWb) ! Ratio of upward to forward diffuse fluxes
+    real(r8) :: Dif_dn(nclmax,maxpft,nlevleaf)           ! Forward diffuse flux onto canopy layer J (W/m**2 ground area)
+    real(r8) :: Dif_up(nclmax,maxpft,nlevleaf)           ! Upward diffuse flux above canopy layer J (W/m**2 ground area)
+    real(r8) :: lai_change(nclmax,maxpft,nlevleaf)       ! Forward diffuse flux onto canopy layer J (W/m**2 ground area)
+    real(r8) :: f_not_abs(maxpft,maxSWb)                   ! Fraction reflected + transmitted. 1-absorbtion.
+    real(r8) :: Abs_dir_z(maxpft,nlevleaf)
+    real(r8) :: Abs_dif_z(maxpft,nlevleaf)
+    real(r8) :: abs_rad(maxSWb)                               !radiation absorbed by soil
+    real(r8) :: tr_soili                                      ! Radiation transmitted to the soil surface.
+    real(r8) :: tr_soild                                      ! Radiation transmitted to the soil surface.
+    real(r8) :: phi1b(maxpft)                                 ! Radiation transmitted to the soil surface.
+    real(r8) :: phi2b(maxpft)
+    real(r8) :: laisum                                        ! cumulative lai+sai for canopy layer (at middle of layer)
+    real(r8) :: angle
+    
+    real(r8),parameter :: tolerance = 0.000000001_r8
+    real(r8), parameter :: pi   = 3.141592654                 ! PI
```

Comment:
ahem

Reply:
?

Reply:
use FatesConstantsMod, only : pi => pi_const?

Reply:
yeah, that would be better, will do